File: sprint5/By lessons/lesson12code2/lesson12code2.cpp
int findKth(Node* root, int k) {
    // Вычисляем размер левого поддерева с учётом того,
    // что оно может быть пустым.
    int leftSize = (root->left == nullptr) ? 0 : root->left->size;
    // Если слева ровно k вершин, то искомая вершина - корень.
    if (leftSize == k) {
        return root->value;
    }
    // Если левое поддерево слишком мало, то продолжим
    // поиск в правом поддереве.
    if (leftSize < k) {
        return findKth(root->right, k - leftSize - 1);
    }
    // Иначе продолжим поиск в левом поддереве.
    return findKth(root->left, k);
}

File: sprint5/By lessons/lesson12code2/lesson12code2.swift
func findKth(root: Node?, k: Int) -> Int {
    // Вычисляем размер левого поддерева с учётом того,
    // что оно может быть пустым.
    let leftSize = (root?.left == nil) ? 0 : root!.left!.size
    // Если слева ровно k вершин, то искомая вершина - корень.
    if leftSize == k {
        return root!.value
    }
    // Если левое поддерево слишком мало, то продолжим
    // поиск в правом поддереве.
    if leftSize < k {
        return findKth(root: root?.right, k: k - leftSize - 1)
    }
    // Иначе продолжим поиск в левом поддереве.
    return findKth(root: root?.left, k: k)
}
